XenModel/Actions/VM/GpuAssignAction.cs

```diff
@@ -64,7 +63,7 @@ protected override void Run()
 
             // New added vGPUs haven't opaque_ref
             foreach (var vGpu in vGpus.FindAll(x => x.opaque_ref == null))
-                AddGpu(vm.Connection.Resolve(vGpu.GPU_group), vm.Connection.Resolve(vGpu.type), vGpu.device);
+                AddGpu(vm.Connection.Resolve(vGpu.GPU_group), vm.Connection.Resolve(vGpu.type), vGpu.device != null? vGpu.device:"0");

```

Comment:
You can use `vGpu.device ?? "0"` , which is the same as `vGpu.device != null ? vGpu.device : "0"`, but much more compact.